src/custom/hooks/useRefetchPriceCallback.ts

```diff
@@ -77,31 +75,51 @@ function _isValidOperatorError(error: any): error is OperatorError {
   return error instanceof OperatorError
 }
 
-function _handleUnsupportedToken({
-  chainId,
-  error,
-  addUnsupportedToken
-}: {
-  chainId: ChainId
+interface HandleQuoteErrorParams {
+  quoteData: QuoteInformationObject | FeeQuoteParams
   error: unknown
   addUnsupportedToken: (params: AddGpUnsupportedTokenParams) => void
-}) {
+  clearQuote: (params: ClearQuoteParams) => void
+  setQuoteError: (params: SetQuoteErrorParams) => void
+}
+
+function _handleQuoteError({
+  quoteData,
+  error,
+  addUnsupportedToken,
+  clearQuote,
+  setQuoteError
+}: HandleQuoteErrorParams) {
   if (_isValidOperatorError(error)) {
-    // Unsupported token
-    if (error.type === ApiErrorCodes.UnsupportedToken) {
-      // TODO: will change with introduction of data prop in error responses
-      const unsupportedTokenAddress = error.description.split(' ')[2]
-
-      console.error(`${error.message}: ${error.description} - disabling.`)
-
-      addUnsupportedToken({
-        chainId,
-        address: unsupportedTokenAddress,
-        dateAdded: Date.now()
-      })
-    } else {
-      // some other operator error occurred, log it
-      console.error(error)
+    switch (error.type) {
+      case ApiErrorCodes.UnsupportedToken: {
+        // TODO: will change with introduction of data prop in error responses
+        const unsupportedTokenAddress = error.description.split(' ')[2]
+        console.error(`${error.message}: ${error.description} - disabling.`)
+
+        return addUnsupportedToken({
+          chainId: quoteData.chainId,
+          address: unsupportedTokenAddress,
+          dateAdded: Date.now()
+        })
+      }
+      // Fee/Price query returns error
+      // e.g Insufficient Liquidity or Fee exceeds Price
+      case ApiErrorCodes.FeeExceedsFrom:
+      case ApiErrorCodes.NotFound: {
+        console.error(`${error.message}: ${error.description}!`)
+        return setQuoteError({
+          ...quoteData,
+          lastCheck: Date.now(),
+          error: error.type
```

Comment:
This connect with one comment i did in the PR where I saw you were using here the same error codes as the backend

I think that we should use our own enum, with just the types we handle. Or a string if we can get by with that.
This way, we make sure we translate from backend errors to frontend errors.

Reply:
ok then i will do that then

Reply:
notes in https://github.com/gnosis/cowswap/issues/751